src/lexer.jl

```diff
@@ -68,16 +68,16 @@ function Base.start(l::Lexer)
     l.current_row = 1
     l.current_col = 1
     l.current_pos = l.io_startpos
-    false
+    t = next_token(l)
+    return t, t.kind == Tokens.ENDMARKER
 end
 
-function Base.next(l::Lexer, ::Any)
+function Base.iterate(l::Lexer, isdone::Any)
+    isdone && return nothing
     t = next_token(l)
     return t, t.kind == Tokens.ENDMARKER
```

Comment:
We could probably do without `Tokens.ENDMARKER` now, but that's a separate concern.